### PR TITLE
Add ai03/soyuz VIA Support

### PR DESCRIPTION
**Pull request by mrsendyyk**
## Description

Add `ai03/soyuz` VIA Support

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [x] New feature
- [x] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* [Add VIA support for Soyuz #314](https://github.com/the-via/keyboards/pull/314)

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/ai03/soyuz/config.h [hunk 1]
```diff
@@ -21,11 +21,10 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
 /* USB Device descriptor parameter */
 #define VENDOR_ID       0xA103
-#define PRODUCT_ID      0x0004
+#define PRODUCT_ID      0x0004 // ai03
```
Comment (drashna) [inline keyboards/ai03/soyuz/config.h:24]: Ummmm ... vender ID?

**Review by noroadsleft — CHANGES_REQUESTED** [review]
This actually won't work because `0xA103:0x0004` is the same Vendor ID/Product ID pair that the Equinox has, and the Equinox is already supported by VIA.

@ai03-2725, what would you like done here? Soyuz was added to QMK first, but Equinox already has VIA support, so Soyuz would be easier to change.

File: keyboards/ai03/soyuz/config.h (hunk 1 above)
Comment (mrsendyyk) [reply keyboards/ai03/soyuz/config.h:24]: @ai03-2725

**Comment by ai03-2725** [comment]
> This actually won't work because `0xA103:0x0004` is the same Vendor ID/Product ID pair that the Equinox has, and the Equinox is already supported by VIA.
> 
> @ai03-2725, what would you like done here? Soyuz was added to QMK first, but Equinox already has VIA support, so Soyuz would be easier to change.

If it is fine, would it be possible to change the product ID of Soyuz from 0004 to 0018?
Honestly no clue how the conflicting ID happened since I usually refer to a running list of PIDs, adding new boards with incremented ID as necessary. Perhaps this board was from before I started keeping track.

**Comment by noroadsleft** [comment]
> If it is fine, would it be possible to change the product ID of Soyuz from 0004 to 0018?
> Honestly no clue how the conflicting ID happened since I usually refer to a running list of PIDs, adding new boards with incremented ID as necessary. Perhaps this board was from before I started keeping track.

Sounds good, thanks.

File: keyboards/ai03/soyuz/config.h (hunk 1 above)
Comment (noroadsleft) [reply keyboards/ai03/soyuz/config.h:24]:
```suggestion
#define PRODUCT_ID      0x0018
```

**Review by noroadsleft — CHANGES_REQUESTED** [review]

**Review by noroadsleft — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

**Comment by noroadsleft** [comment]
Thanks!

**Pull request merged**